File: matrix_rotation/matrix.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int case_num;
    scanf("%d", &case_num);
    for (int i = 0; i < case_num; i++)
    {
        int x_1, x_2, x_3, x_4;
        scanf("%d%d%d%d", &x_1, &x_2, &x_3, &x_4);
        if (((x_1 < x_2 && x_3 < x_4) || (x_1 > x_2 && x_3 > x_4)) && ((x_1 < x_3 && x_2 < x_4) || (x_1 > x_3 && x_2 > x_4)))
        {
            printf("YES\n");
        }

        else
        {
            printf("NO\n");
        }
    }
}

File: matrix_rotation/matrix.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        x_1, x_2 = input().split()
        x_3, x_4 = input().split()
        x_1 = int(x_1)
        x_2 = int(x_2)
        x_3 = int(x_3)
        x_4 = int(x_4)
        if ((((x_1 < x_2 and x_3 < x_4) or (x_1 > x_2 and x_3 > x_4)) and ((x_1 < x_3 and x_2 < x_4) or (x_1 > x_3 and x_2 > x_4)))):
            print("YES")
        else:
            print("NO")
    
    
if __name__ == "__main__":
    main()